### PR TITLE
feat(settings/native): start car mode in low bandwidth mode

### PR DESCRIPTION
**Pull request by Calinteodor**
1. Added 'Start car mode in low bandwidth mode' switch to Settings screen.
2. And the switch is checked and 'Join in low bandwidth mode' button inside Prejoin screen is pressed, the participant goes straight to Carmode screen.

### CODE REVIEW

File: react/features/prejoin/components/Prejoin.native.tsx [hunk 1]
```diff
@@ -85,7 +84,12 @@ const Prejoin: React.FC<PrejoinProps> = ({ navigation }: PrejoinProps) => {
 
     const onJoinLowBandwidth = useCallback(() => {
         dispatch(setAudioOnly(true));
-        onJoin();
+        dispatch(connect());
+        if (startCarmode) {
+            navigateRoot(screen.conference.carmode);
```
Comment (saghul) [inline react/features/prejoin/components/Prejoin.native.tsx:89]: This is not what we want, we want to go into the conference, and THEN enter car mode, so the user can then close it if they want to.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/conference/components/native/carmode/CarMode.tsx [hunk 2]
```diff
@@ -26,16 +33,40 @@ import TitleBar from './TitleBar';
...
 import styles from './styles';
 
+
 /**
  * Implements the carmode component.
  *
  * @returns { JSX.Element} - The carmode component.
  */
-const CarMode = (): JSX.Element => {
+const CarMode = ({ navigation }: any): JSX.Element => {
```
Comment (saghul) [inline react/features/conference/components/native/carmode/CarMode.tsx:42]: I don't think we need any of these changes anymore.

File: react/features/toolbox/components/native/AudioOnlyButton.js [hunk 3]
```diff
@@ -41,9 +50,19 @@ class AudioOnlyButton extends AbstractButton<Props, *> {
      * @returns {void}
      */
     _handleClick() {
-        this.props.dispatch(toggleAudioOnly());
+        const { _audioOnly, _startCarMode, dispatch } = this.props;
+
+        if (!_audioOnly && _startCarMode) {
+            dispatch(setAudioOnly(true));
+            navigate(screen.conference.carmode);
+        } else if (_audioOnly && _startCarMode) {
```
Comment (saghul) [inline react/features/toolbox/components/native/AudioOnlyButton.js:58]: I think you can safely drop this branch and just have the else.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/toolbox/components/native/AudioOnlyButton.js (hunk 3 above)
Comment (Calinteodor) [reply react/features/toolbox/components/native/AudioOnlyButton.js:58]: Tested it and works. Will update.

File: react/features/conference/components/native/carmode/CarMode.tsx (hunk 2 above)
Comment (Calinteodor) [reply react/features/conference/components/native/carmode/CarMode.tsx:42]: Reverted.

**Review by saghul — APPROVED** [review]

**Pull request merged**